Add "Created On" column to transfer list

aaron-kruse opened the pull request:
Makes the torrent's "Created On" date/time available (but hidden by default) in the main transfer list for sorting and better visibility.

Unlike the "Added On" column (which shows when a torrent was added to the client's list), this shows the date the torrent file was originally created. This is something that can help prioritize which torrents to seed. I have a long list of torrents and looking at each one's properties for the "Created On" date isn't feasible. With the addition of this column, I can just sort by it and bring the oldest torrents to the top of the list. Similarly, I can sort by the number of seeders to find which torrents could use some help the most, then glance at the "Created On" column (ex. I'd rather seed torrents from 10+ years ago that are struggling due to age/availability vs. ones from a few weeks ago that have been replaced by newer/better options). This column could also help with cleaning up your list of torrents by making them easier to work with based on their actual age (and not date added/completed).

I should also maybe mention that this is my first submission here, so hopefully I've followed all the guidelines correctly. I added this column using other similar columns as an example and everything seems to be working fine, but if I missed anything less obvious, please let me know!

Screenshot for reference:
<img width="1496" height="723" alt="image" src="https://github.com/user-attachments/assets/e009692c-73b2-4ed9-bfe1-2e939a541f97" />

aaron-kruse committed:
Add "Created On" column to transfer list

This column makes the torrent's "Created On" date/time available in the
main transfer list for sorting and better visibility.

stalkerok commented:
- https://github.com/qbittorrent/qBittorrent/issues/13321

aaron-kruse commented:
I saw this was approved a little while ago but hasn't been merged yet. Is there anything else I need to do here?

Any rough idea when this might be included in a release? I'm not trying to be pushy or anything, I just don't know what the release process/timeline is like and am curious when this might be included in an official release (ex. if it might be soonish/later this month, or if this will still be a few months out).

glassez commented:
https://github.com/qbittorrent/qBittorrent/pull/23647#discussion_r2649246469 isn't resolved yet.
@Chocobo1, ping!

aaron-kruse committed:
Move new enum member to end of list

Workaround to avoid issues with existing client column config that is dependent on the current ordering of members described here: https://github.com/qbittorrent/qBittorrent/pull/23647#discussion_r2701417821

aaron-kruse commented on src/gui/transferlistmodel.h:
```diff
@@ -65,6 +65,7 @@ class TransferListModel final : public QAbstractListModel
         TR_POPULARITY,
         TR_CATEGORY,
         TR_TAGS,
+		TR_CREATE_DATE,
```
Done in c73e36994e62fd40570f926e696d0f65959b3bf6.
